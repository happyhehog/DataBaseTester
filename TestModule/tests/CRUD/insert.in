[@Clear]
[@PrintLevel] MAIN

@@ ==============
@@ Testing INSERT
@@ ==============

CREATE TABLE t(id INT PRIMARY KEY NOT NULL, height FLOAT, nickname CHAR(50));

SHOW CREATE TABLE t;

INSERT INTO t VALUES (1, 170.0, 'hello');
INSERT INTO t(id, height, nickname) VALUES (2, 180.0, 'world');
INSERT INTO t(id) VALUES (3);
INSERT INTO t(id, height) VALUES(4, 120.3);
INSERT INTO t(id, height) VALUES(5, 110.3);
INSERT INTO t(id, height) VALUES(6, 120.3);
INSERT INTO t(id, height) VALUES (7, 150.0);
INSERT INTO t(id, height) VALUES (8, 120.0);
INSERT INTO t(id, height) VALUES (9, 120);
INSERT INTO t VALUES (10, 400.0, 'poilkj');
INSERT INTO t VALUES (11, 10.9, 'qqqwww');
INSERT INTO t(id, height) VALUES (12, 10);
INSERT INTO t(height, nickname, id) VALUES(20.0, 'asdfkljasdf', 13);

@@ Иногда вылетает TimeOut
INSERT INTO t(id, nickname) VALUES (14, 'eoeoe');

@@ Иногда вылетает TimeOut
INSERT INTO t(id, nickname) VALUES (15, 'qweqwe');

@@ Ожидается отсутствие TimeOut, но рандом
INSERT INTO t VALUES (16, 170.0, 'hello');

[@Print]
SELECT * FROM t;
DROP TABLE t;


[@Print]
[@Clear]
[@RestartServer]
[@NoOutput]
CREATE TABLE t(id INT PRIMARY KEY NOT NULL, height FLOAT, nickname CHAR(50));
[@NoOutput]

@@ Гарантировать отсутствие TimeOut может только очистка и перезапуск сервера
INSERT INTO t VALUES (16, 170.0, 'hello');

[@NoOutput]
DROP TABLE t;
[@NoOutput]