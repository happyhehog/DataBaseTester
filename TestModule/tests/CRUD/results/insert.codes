{"status":"OK","command":"CREATE TABLE t(id INT PRIMARY KEY NOT NULL, height FLOAT, nickname CHAR(50));","answer":"Success"}
{"status":"OK","command":"SHOW CREATE TABLE t;","answer":"CREATE TABLE t(id INT PRIMARY KEY NOT NULL, height FLOAT, nickname CHAR(50));"}
{"status":"OK","command":"INSERT INTO t VALUES (1, 170.0, 'hello');","answer":"Success"}
{"status":"OK","command":"INSERT INTO t(id, height, nickname) VALUES (2, 180.0, 'world');","answer":"Success"}
{"status":"OK","command":"INSERT INTO t(id) VALUES (3);","answer":"Success"}
{"status":"OK","command":"INSERT INTO t(id, height) VALUES(4, 120.3);","answer":"Success"}
{"status":"OK","command":"INSERT INTO t(id, height) VALUES(5, 110.3);","answer":"Success"}
{"status":"OK","command":"INSERT INTO t(id, height) VALUES(6, 120.3);","answer":"Success"}
{"status":"OK","command":"INSERT INTO t(id, height) VALUES (7, 150.0);","answer":"Success"}
{"status":"OK","command":"INSERT INTO t(id, height) VALUES (8, 120.0);","answer":"Success"}
{"status":"OK","command":"INSERT INTO t(id, height) VALUES (9, 120);","answer":"Success"}
{"status":"OK","command":"INSERT INTO t VALUES (10, 400.0, 'poilkj');","answer":"Success"}
{"status":"OK","command":"INSERT INTO t VALUES (11, 10.9, 'qqqwww');","answer":"Success"}
{"status":"OK","command":"INSERT INTO t(id, height) VALUES (12, 10);","answer":"Success"}
{"status":"OK","command":"INSERT INTO t(height, nickname, id) VALUES(20.0, 'asdfkljasdf', 13);","answer":"Success"}
{"status":"OK","command":"INSERT INTO t(id, nickname) VALUES (14, 'eoeoe');","answer":"Success"}
{"status":"OK","command":"INSERT INTO t(id, nickname) VALUES (15, 'qweqwe');","answer":"Success"}
{"status":"OK","command":"INSERT INTO t VALUES (16, 170.0, 'hello');","answer":"Success"}
{"status":"OK","command":"SELECT * FROM t;","answer":"id|height    |nickname     |\n1 |170.000000|'hello'      |\n2 |180.000000|'world'      |\n3 |          |             |\n4 |120.300000|             |\n5 |110.300000|             |\n6 |120.300000|             |\n7 |150.000000|             |\n8 |120.000000|             |\n9 |120.000000|             |\n10|400.000000|'poilkj'     |\n11|10.900000 |'qqqwww'     |\n12|10.000000 |             |\n13|20.000000 |'asdfkljasdf'|\n14|          |'eoeoe'      |\n15|          |'qweqwe'     |\n16|170.000000|'hello'      |"}
{"status":"OK","command":"DROP TABLE t;","answer":"Success"}
{"status":"OK","command":"INSERT INTO t VALUES (16, 170.0, 'hello');","answer":"Success"}
