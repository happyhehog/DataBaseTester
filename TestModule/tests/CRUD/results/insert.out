Success
CREATE TABLE t(id INT PRIMARY KEY NOT NULL, height FLOAT, nickname CHAR(50));
Success
Success
Success
Success
Success
Success
Success
Success
Success
Success
Success
Success
Success
Success
Success
Success

id|height    |nickname     |
1 |170.000000|'hello'      |
2 |180.000000|'world'      |
3 |          |             |
4 |120.300000|             |
5 |110.300000|             |
6 |120.300000|             |
7 |150.000000|             |
8 |120.000000|             |
9 |120.000000|             |
10|400.000000|'poilkj'     |
11|10.900000 |'qqqwww'     |
12|10.000000 |             |
13|20.000000 |'asdfkljasdf'|
14|          |'eoeoe'      |
15|          |'qweqwe'     |
16|170.000000|'hello'      |
Success

Success
